x-pack/legacy/plugins/apm/public/components/shared/DottedKeyValueTable/index.tsx

```diff
@@ -5,75 +5,27 @@
  */
 
 import React, { TableHTMLAttributes } from 'react';
-import { compact, isObject } from 'lodash';
 import {
   EuiTable,
   EuiTableProps,
   EuiTableBody,
   EuiTableRow,
   EuiTableRowCell
 } from '@elastic/eui';
-import { StringMap } from '../../../../typings/common';
 import { FormattedValue } from './FormattedValue';
-
-interface PathifyOptions {
-  maxDepth?: number;
-  parentKey?: string;
-  depth?: number;
-}
-
-interface PathifyResult {
-  [key: string]: any;
-}
-
-/**
- * Converts a deeply-nested object into a one-level object
- * with dot-notation paths as keys.
- */
-export function pathify(
-  item: StringMap<any>,
-  { maxDepth, parentKey = '', depth = 0 }: PathifyOptions
-): PathifyResult {
-  const isArrayWithSingleValue = Array.isArray(item) && item.length === 1;
-  return Object.keys(item)
-    .sort()
-    .reduce((pathified, key) => {
-      const childKey = isArrayWithSingleValue ? '' : key;
-      const currentKey = compact([parentKey, childKey]).join('.');
-      if ((!maxDepth || depth + 1 <= maxDepth) && isObject(item[key])) {
-        return {
-          ...pathified,
-          ...pathify(item[key], {
-            maxDepth,
-            parentKey: currentKey,
-            depth: depth + 1
-          })
-        };
-      } else {
-        return { ...pathified, [currentKey]: item[key] };
-      }
-    }, {});
-}
+import { KeyValuePair } from '../../../utils/flattenObject';
 
 export function DottedKeyValueTable({
-  data,
-  parentKey,
-  maxDepth,
+  items,
```

Comment:
What about naming this "rows"? That fits nicely with the "table" terminology.
```suggestion
  rows,
```